src/Nest/Indices/MappingManagement/GetMapping/GetMappingResponse.cs

```diff
@@ -44,6 +44,13 @@ public class GetMappingResponse : DictionaryResponseBase<IndexName, IndexMapping
 		[JsonIgnore]
 		public IReadOnlyDictionary<IndexName, IndexMappings> Mappings => Indices;
 
+		[Obsolete("Use GetMappingFor explicitly instead this is a leaky abstraction that returns the mapping of the first index's first type on the response")]
+		public TypeMapping Mapping => this.Indices.FirstOrDefault().Value?.Mappings?.FirstOrDefault().Value;
+
+		public TypeMapping GetMappingFor<T>() => this.Indices[typeof(T)]?[typeof(T)];
+		public TypeMapping GetMappingFor(string index, string type) => this.Indices[index]?[type];
```

Comment:
Should any argument validation checks happen here?

Reply:
On second thought, should this return ITypeMapping?

Reply:
++ this could be a whole lot more defensive 👍